tests/acceptance/17_packages/10_new/unsafe/the_great_package_test_generator.py

```diff
@@ -0,0 +1,542 @@
+#!/usr/bin/python
+
+from __future__ import print_function
+
+import pprint
+import sys
+
+TEST_COUNT = 0
+PLATFORM_ARCH_32 = False
+
+states = {
```

Comment:
I'd be inclined to generate this programmatically:
```
def states(top=2, archs=("64_bit", "32_bit")):
    """Iterate the possible states.

    Here, state 0 means absent, which won't be included in the state description.
    The other two states are different versions (1 and 2) of a package.
    Each arch can be absent or at either version.
    If both are present, they are constrained to have the same version.
    We iterate pairs of a state-name and a state-description dictionary.
    The dictionary's keys are (arch, version) pairs, mapped to True if
    the given arch is present at the specified version.
    """
    # Empty state - pair each arch with each version (not 0) and map the pair to False:
    blank = dict(((arch, str(ver)), False) for arch in archs for ver in range(1, top + 1))
    # Each conceivable state is expressed as a value for this list:
    vers = [top for v in archs] # as many top as entries in archs
    while True:
        if len(filter(None, set(vers))) < 2: # we have at most one non-zero version
            pairs = [(arch, str(ver)) for arch, ver in zip(archs, vers) if ver]
            yield ('_'.join('_'.joiin(p) for p in pairs) or 'absent',
                   dict((pair, pair in pairs) for pair in blank.items()))

        # Now step vers to the next conceivable (not necessarily valid) candidate:
        ind = len(vers)
        while ind > 0:
            ind -= 1
            if vers[ind] > 0:
                vers[ind] -= 1
                break
            else:
                vers[ind] = top
        else: break # we fell off the bottom, ind < 0 so we have finished.

# Now collect those up in a dictionary mapping state name to its details:
states = dict(states())
```
(Note that the `else` block of a `while` or `for` loop is run precisely if we don't `break` out of the loop.
The function zip() is a transposition: zip(('a', 'b', 'c'), (1, 2, 3)) is (('a', 1), ('b', 2), ('c', 3)) and similar)
Unfortunately, I don't think that's very clear to the reader !